pvanalytics/quality/gaps.py

```diff
@@ -135,3 +136,206 @@ def interpolation_diff(x, window=3, rtol=1e-5, atol=1e-8):
+    the day for which there is data (a value other than NaN). The time
+    attributed to each value is equal to the timestamp spacing of
+    `series` or `freq` if it is specified. For example, a day with 24
+    non-NaN values in a series with 30 minute timestamp spacing would
+    have 12 hours of data and therefore completeness score of 0.5.
+
+    Parameters
+    ----------
+    series : Series
+        A DatetimeIndexed series.
+    freq : str, default None
+        Interval between samples in the series as a pandas frequency
+        string. If None, the frequency is inferred using
+        :py:func:`pandas.infer_freq`.
+
+    Returns
+    -------
+    Series
+        A series of floats, indexed by day, giving the completeness
+        score for each day (fraction of the day for which `series` has
+        data).
+
+    Raises
+    ------
+    ValueError
+        If `freq` is longer than the frequency inferred from `series`.
+
+    """
+    inferred_seconds = _freq_to_seconds(pd.infer_freq(series.index))
+    freq_seconds = _freq_to_seconds(freq)
+    seconds_per_sample = freq_seconds or inferred_seconds
+    if freq and inferred_seconds < freq_seconds:
+        raise ValueError("freq must be less than or equal to the"
+                         + " frequency of the series")
+    daily_counts = series.resample('D').count()
+    return (daily_counts * seconds_per_sample) / (1440*60)
+
+
+def complete(series, minimum_completeness=0.333, freq=None):
+    """Select only data points that are part of a day with complete data.
+
+    A day is complete if its completeness score is greater than or
+    equal to `minimum_completeness`. See :py:func:`daily_completeness` for more
+    information. For example, a day with 24 non-NaN values in a series
+    with 30 minute timestamp spacing would have 12 hours of data and
+    therefore a completeness score of 0.5; with the default
+    `minimum_completeness=0.333` the day would be marked complete.
+
+    Parameters
+    ----------
+    series : Series
+        The data to be checked for completeness.
+    minimum_completeness : float, default 0.333
+        Fraction of the day that must have data.
+    freq : str, default None
+        The expected frequency of the data in `series`. If none then
+        the frequency is inferred from the data.
+
+    Returns
+    -------
+    Series
+        A series of booleans with True for each value that is part of
+        a day with completeness greater than `minimum_completeness`.
+
+    Raises
+    ------
+    ValueError
+        See :py:func:`daily_completeness`.
+
+    See Also
+    --------
+    :py:func:`daily_completeness`
+
+    """
+    complete_days = daily_completeness(series, freq) >= minimum_completeness
+    return complete_days.reindex(series.index, method='pad')
+
+
+def start_stop_dates(series, days=10, minimum_completeness=0.333333,
+                     freq=None):
+    """Get the start and end of data excluding leading and trailing gaps.
```

Comment:
Perhaps this function could be made more general by accepting a time series of Boolean, and returning start and stop where the minimum daily count of True is less than the threshold.  That way the criteria for start and stop could change, e.g., fraction of times with irradiance like clear sky.

Reply:
If we want to keep this function as is, for PVFleets, we could make a `trim_pvfleets` function that implements their workflow.

Reply:
I like the idea of making this more general is good. If you think passing a boolean series in would achieve that, then I will go ahead and make that change. `trim_pvfleets` (or whatever we want to call it) can easily be implemented using the more general function.